src/nectarchain/dqm/pixel_participation.py

```diff
@@ -0,0 +1,148 @@
+from dqm_summary_processor import DQMSummary
+from matplotlib import pyplot as plt
+from ctapipe.visualization import CameraDisplay
+from ctapipe.instrument import CameraGeometry
+from ctapipe.coordinates import EngineeringCameraFrame
+import numpy as np
+
+
+class PixelParticipation_HighLowGain(DQMSummary):
+    def __init__(self, gaink):
+        self.k = gaink
+        return None
+
+    def ConfigureForRun(self, path, Pix, Samp, Reader1):
+        # define number of pixels and samples
+        self.Pix = Pix
+        self.Samp = Samp
+
+        self.counter_evt = 0
+        self.counter_ped = 0
+
+        self.camera = CameraGeometry.from_name("NectarCam-003").transform_to(EngineeringCameraFrame())
+        self.camera2 = CameraGeometry.from_name("NectarCam-003").transform_to(EngineeringCameraFrame())
+
+        self.cmap = "gnuplot2"
+        self.cmap2 = "gnuplot2"
+
+
+        self.BadPixels_ped = np.zeros(self.Pix)
+        self.BadPixels =  np.zeros(self.Pix)
+
+    def ProcessEvent(self, evt, noped):
+        pixelBAD = evt.mon.tel[0].pixel_status.hardware_failing_pixels[self.k]
+        pixel = evt.nectarcam.tel[0].svc.pixel_ids
+        if len(pixel) < self.Pix:
+            pixel21 = list(np.arange(0, self.Pix - len(pixel), 1, dtype=int))
+            pixel = list(pixel)
+            pixels = np.concatenate([pixel21, pixel])
+        else: 
+            pixels = pixel
+
+        if evt.trigger.event_type.value == 32:  # count peds
+            self.counter_ped += 1
+            BadPixels_ped1 = list(map(int, pixelBAD[pixels]))
+            self.BadPixels_ped += BadPixels_ped1
+
+        else:
+            self.counter_evt += 1
+            BadPixels1 = list(map(int, pixelBAD[pixels]))
+            self.BadPixels += BadPixels1
+        return None
+
+    def FinishRun(self):
+        self.BadPixels_ped = np.array(self.BadPixels_ped)
+        self.BadPixels = np.array(self.BadPixels)
+
+
+
+    def GetResults(self):
```

Comment:
Not PEP-8 compliant. Did you enable the `pre-commit` hooks before your commit ?

Reply:
enabled now